Update core doc (#13324)
openvinotoolkit/openvino: src/core/docs/debug_capabilities.md

# OpenVINO Core debug capabilities

> OpenVINO Core contains a set of different debug capabilities which allow making developer life a little bit easier, because these capabilities support to collect information about object statuses in the process of OpenVINO Runtime execution and report this information to developer.

Reviewer: ```suggestion
OpenVINO Core contains a set of different debug capabilities that make developer life easier by collecting information about object statuses during OpenVINO Runtime execution and reporting this information to the developer.
```

# OpenVINO Core debug capabilities

> OpenVINO Core contains a set of different debug capabilities which allow to make developer live a little bit easier.

Reviewer: allow to make ---> allow making
 live  ---> life 
 overall pretty useless sentence, can we elaborate on how exactly it makes it easier or just remove it?

Author: Changed sentence.
